controllers/wso2is_controller.go

```diff
@@ -0,0 +1,592 @@
+				"nginx.ingress.kubernetes.io/backend-protocol":    "HTTPS",
+				"nginx.ingress.kubernetes.io/affinity":            "cookie",
+				"nginx.ingress.kubernetes.io/session-cookie-name": "route",
+				"nginx.ingress.kubernetes.io/session-cookie-hash": "sha1",
+			},
+		},
+		Spec: v1beta1.IngressSpec{
+			TLS: []v1beta1.IngressTLS{
+				{
+					Hosts: []string{m.Spec.Configurations.Host},
+				},
+			},
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: m.Spec.Configurations.Host,
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{{
+								Path: "/",
+								Backend: v1beta1.IngressBackend{
+									ServiceName: "wso2is-service",
+									ServicePort: intstr.IntOrString{
+										IntVal: 9443,
+									},
+								},
+							}},
+						},
+					},
+				},
+			},
+		},
+	}
+	ctrl.SetControllerReference(&m, ingress, r.Scheme)
+	return ingress
+}
+
+// addNewService adds a new Service
+func (r *Wso2IsReconciler) addNewService(m wso2v1beta1.Wso2Is) *corev1.Service {
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "wso2is-service",
+			Namespace: m.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{
+				Name:     "servlet-http",
+				Protocol: "TCP",
+				Port:     9763,
+				TargetPort: intstr.IntOrString{
+					IntVal: 9763,
+				},
+			}, {
+				Name:     "servlet-https",
+				Protocol: "TCP",
+				Port:     9443,
+				TargetPort: intstr.IntOrString{
+					IntVal: 9443,
+				},
+			}},
+			Selector: labelsForWso2IS(m.Name),
+			Type:     corev1.ServiceTypeLoadBalancer,
+		},
+	}
+	ctrl.SetControllerReference(&m, svc, r.Scheme)
+	return svc
+}
+
+// New deployment for WSO2IS
+func (r *Wso2IsReconciler) deploymentForWso2Is(m wso2v1beta1.Wso2Is) *appsv1.Deployment {
+	ls := labelsForWso2IS(m.Name)
+	replicas := m.Spec.Size
+	runasuser := int64(802)
+
+	dep := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      m.Name,
+			Namespace: m.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: ls,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: ls,
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "user-store-persistent-storage",
+							VolumeSource: corev1.VolumeSource{
+								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+									ClaimName: "user-store-pv-claim",
+								},
+							},
+						},
+						{
+							Name: "identity-server-conf",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "identity-server-conf",
+									},
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{{
+						Name:  "wso2is",
+						Image: "sureshmichael/wso2-is-5.11.0:rc1",
```

Comment:
Don't we need to make this configurable?

Reply:
To be decided. Depending on how we are going to rollout I guess